packages/discovery-provider/integration_tests/queries/test_get_attestation.py

```diff
@@ -123,6 +126,149 @@ def test_get_attestation(app):
                 )
 
 
+def test_get_attestation_weekly_pool_exhausted(app):
+    with app.app_context():
+        db = get_db()
+
+    entities = {
+        "users": [
+            {"user_id": 1, "wallet": "0x38C68fF3926bf4E68289672F75ee1543117dDAAA"},
+            {"user_id": 2, "wallet": "0x38C68fF3926bf4E68289672F75ee1543117dD9B3"},
+            {"user_id": 3, "wallet": "0xDEADBEEFDEADBEEFDEADBEEFDEADBEEFDEADBEEF"},
+        ],
+        "challenges": [
+            {
+                "id": "not-exhausted",
+                "type": ChallengeType.numeric,
+                "active": True,
+                "amount": 5,
+                "weekly_pool": 10,
+            },
+            {
+                "id": "exhausted",
+                "type": ChallengeType.numeric,
+                "active": True,
+                "amount": 10,
+                "weekly_pool": 20,
+            },
+        ],
+        "user_challenges": [
+            {
+                "challenge_id": "not-exhausted",
+                "user_id": 1,
+                "specifier": "1",
+                "is_complete": True,
+                "amount": 5,
+            },
+            {
+                "challenge_id": "not-exhausted",
+                "user_id": 2,
+                "specifier": "2",
+                "is_complete": True,
+                "amount": 5,
+            },
+            {
+                "challenge_id": "exhausted",
+                "user_id": 1,
+                "specifier": "1",
+                "is_complete": True,
+                "amount": 10,
+            },
+            {
+                "challenge_id": "exhausted",
+                "user_id": 2,
+                "specifier": "2",
+                "is_complete": True,
+                "amount": 10,
+            },
+            # User 3 has completed challenge `exhausted` but will not be able to disburse
+            {
+                "challenge_id": "exhausted",
+                "user_id": 3,
+                "specifier": "3",
+                "is_complete": True,
+                "amount": 10,
+            },
+        ],
+        "challenge_disbursements": [
+            {
+                "challenge_id": "not-exhausted",
+                "specifier": "1",
+                "user_id": 1,
+                "amount": "5",
+            },
+            {
+                "challenge_id": "exhausted",
+                "specifier": "1",
+                "user_id": 1,
+                "amount": "10",
+            },
+            {
+                "challenge_id": "exhausted",
+                "specifier": "2",
+                "user_id": 2,
+                "amount": "10",
+            },
+        ],
+    }
+    populate_mock_db(db, entities)
+    with db.scoped_session() as session:
+        oracle_address = "0x32a10e91820fd10366AC363eD0DEa40B2e598D22"
```

Comment:
Magic string? Does this match with a constant somewhere else?

Reply:
Just stole this all from the prev test, but added a const for it 👍